x/profiles/module_ibc.go

```diff
@@ -149,28 +184,93 @@ func (am AppModule) OnRecvPacket(
 			sdk.NewAttribute(types.AttributeChainLinkSourceAddress, address.GetAddress()),
 			sdk.NewAttribute(types.AttributeChainLinkSourceChainName, packetData.SourceChainConfig.Name),
 			sdk.NewAttribute(types.AttributeChainLinkDestinationAddress, packetData.DestinationAddress),
-			sdk.NewAttribute(types.AttributeKeyAckSuccess, fmt.Sprintf("%t", err != nil)),
+			sdk.NewAttribute(types.AttributeKeyAckSuccess, fmt.Sprintf("%t", true)),
 		),
 	)
 
-	// Encode acknowledgement
-	ackBytes, err := sdk.SortJSON(types.ProtoCdc.MustMarshalJSON(&ack))
+	return acknowledgement, nil
+}
+
+// handleOracleRequestPacketData tries handling a OracleResponsePacketData packet.
+// Returns ErrInvalidPacketData if the given packet data is not of such type.
+func (am AppModule) handleOracleRequestPacketData(
+	ctx sdk.Context, packet channeltypes.Packet,
+) (channeltypes.Acknowledgement, error) {
+	var data oracletypes.OracleResponsePacketData
+	if err := oracletypes.ModuleCdc.UnmarshalJSON(packet.GetData(), &data); err != nil {
+		return channeltypes.Acknowledgement{}, sdkerrors.Wrapf(types.ErrInvalidPacketData, "%T", packet)
+	}
+
+	acknowledgement := channeltypes.NewResultAcknowledgement([]byte{byte(1)})
+
+	err := am.keeper.OnRecvApplicationLinkPacketData(ctx, data)
 	if err != nil {
-		return nil, []byte{}, sdkerrors.Wrap(sdkerrors.ErrInvalidType, err.Error())
+		acknowledgement = channeltypes.NewErrorAcknowledgement(err.Error())
 	}
 
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			types.EventTypePacket,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+			sdk.NewAttribute(types.AttributeKeyClientID, data.ClientID),
+			sdk.NewAttribute(types.AttributeKeyRequestID, fmt.Sprintf("%d", data.RequestID)),
+			sdk.NewAttribute(types.AttributeKeyResolveStatus, data.ResolveStatus.String()),
+			sdk.NewAttribute(types.AttributeKeyAckSuccess, fmt.Sprintf("%t", err != nil)),
+		),
+	)
+
 	// NOTE: acknowledgement will be written synchronously during IBC handler execution.
-	return &sdk.Result{
-		Events: ctx.EventManager().Events().ToABCIEvents(),
-	}, ackBytes, nil
+	return acknowledgement, nil
 }
 
 // OnAcknowledgementPacket implements the IBCModule interface
 func (am AppModule) OnAcknowledgementPacket(
 	ctx sdk.Context,
-	modulePacket channeltypes.Packet,
+	packet channeltypes.Packet,
 	acknowledgement []byte,
 ) (*sdk.Result, error) {
+	var ack channeltypes.Acknowledgement
+	if err := types.AminoCdc.UnmarshalJSON(acknowledgement, &ack); err != nil {
```

Comment:
```suggestion
	if err := types.ProtoCdc.UnmarshalJSON(acknowledgement, &ack); err != nil {
```
The codec for packet should be `ProtoCdc`.